src/MongoDB.Driver.Core/Core/Clusters/DnsMonitor.cs

```diff
@@ -161,7 +161,7 @@ private void Monitor()
         {
             while (true)
             {
-                if (_cluster.ShouldDnsMonitorStop())
+                if (_processDnsResultHasEverBeenCalled && _cluster.ShouldDnsMonitorStop())
```

Comment:
Do we need to update the logic in `ShouldDnsMonitorStop()` Currently it reads: 
```
 bool IDnsMonitoringCluster.ShouldDnsMonitorStop()
        {
            var clusterType = Description.Type;
            return clusterType != ClusterType.Unknown && clusterType != ClusterType.Sharded;
        }
```
Should the final line be`return clusterType != ClusterType.Unknown && clusterType != ClusterType.Sharded && clusterType!= ClusterType.ReplicaSet`

Reply:
No, a not added condition for `ReplicaSet` is expected by the spec. The only problem which we have that we need to call host resolving logic at least first time. All other attempts should follow the previous rules

Reply:
No, that wouldn't be correct. If the clusterType  != ReplicaSet, we _don't_ want to stop monitoring, we want to keep monitoring.   We _could_ add another conditional here, like:

      return (clusterType != ClusterType.Unknown && clusterType != ClusterType.Sharded_ && (isInitialized)

but that would end up being redundant to the extra check we have added here.

Reply:
Gotcha, thanks for the explanation!

Reply:
And we don't have a convenient way to check here for whether it's been initialized already